packages/experience/src/pages/ForgotPassword/ForgotPasswordForm/index.tsx

```diff
@@ -70,10 +73,16 @@ const ForgotPasswordForm = ({
           return;
         }
 
+        /**
+         * Update user interaction identifier session,
+         * since the identifier may changed in this form
+         */
+        setIdentifierInputValue({ type, value });
```

Comment:
what if the username + password sign-in flow, the user clicks on the forgot-password and uses the email to reset their password?

Reply:
per offline discussion with @Rany0101 , the `forgotPasswordIdentifierInputValue` should be stored separatly.